tests/token_token_create_transaction.rs

```diff
@@ -0,0 +1,306 @@
+
+            // Signing of the transaction
+            match &used_data.private_keys {
+                    Some(private_keys) => {
+                        match keylist.threshold {
+                            // Sign only with the required number of Privatekey according to keylist threshold
+                            Some(threshold) => {
+                                for i in 0..threshold {
+                                    token_create_tx.sign(private_keys[i as usize].clone());
+                                }
+                                ();
+                            }
+                            // Sign with all Privatekeys
+                            None => {
+                                for private_key in private_keys {
+                                    token_create_tx.sign(private_key.clone());
+                                }
+                                ();
+                            }
+                        }
+                    }
+                    None => panic!("Keylist with public keys but no list with private keys to sign the transaction provided."),
+                }
+            ();
+        }
+    }
+    
+    // If token supply is finite a maximum supply must be set
+    if used_data.token_supply_type == TokenSupplyType::Finite {
+        match used_data.max_supply {
+            Some(max_supply) => {
+                token_create_tx.max_supply(max_supply);
+                ();
+            }
+            None => panic!("Token Supply Type is finite but no max supply was provided."),
+        }
+    }
+
+    // Execute transaction with max fee of 100 HBar to be sure it gets executed
+    let token_create_response = match token_create_tx.max_transaction_fee(Hbar::new(100)).execute(client).await {
+        Ok(tx_response) => tx_response,
+        Err(e) => panic!("Token Create Transaction failed with error: {}", e)
+    };
+
+    // Get transaction receipt
+    let receipt = match token_create_response.get_receipt(client).await {
+        Ok(receipt) => receipt,
+        Err(e) => panic!("Transaction Receipt failed with error: {}", e)
+    };
+
+    // Ectract and return only token id
+    match receipt.token_id {
+        Some(token_id) => return token_id,
+        None => panic!("Token Id retrieval failed")
+    };
+}
+
+// Fetch token data on chain with TokenInfo query
+async fn get_token_info(client: &Client, token_id: TokenId) -> TokenInfo{
+    let _ = match TokenInfoQuery::new()
+        .token_id(token_id)
+        .execute(client)
+        .await {
+            Ok(token_info) => return token_info,
+            Err(e) => panic!("Token Info Query failed with error: {}", e)
+        };
+}
+
+// Compare all data from the token onchain (token_info) with the previously used data (used_data)
+fn check_token_info(token_info: TokenInfo, used_data: &UsedData) {
+    assert_eq!(token_info.name, used_data.name, "On chain name does not match name from used_Data.");
```

Comment:
If any of these fail, it's expected that the investigation would start by checking the line numbers, so, the comments aren't all that useful